controllers/horizon_controller.go

```diff
@@ -340,6 +344,29 @@ func (r *HorizonReconciler) reconcileNormal(ctx context.Context, instance *horiz
 		return ctrl.Result{}, err
 	}
 
+	// Create Memcached instance if no shared instance exists.
+	if instance.Spec.SharedMemcached == "" {
+		memcached := r.renderMemcached(instance)
+		op, err := controllerutil.CreateOrPatch(ctx, helper.GetClient(), memcached, func() error {
+			memcached.Labels = map[string]string{"service": instance.Name}
+			err := controllerutil.SetControllerReference(helper.GetBeforeObject(), memcached, helper.GetScheme())
```

Comment:
Adding a label here is probably unnecessary since we have the owner reference set as well. Fairly easy to tell which service has created the memcached instance by just checking the `ownerReferences`:

```
❯ oc get memcached horizon-memcached -o jsonpath='{.metadata.ownerReferences}' | jq
[
  {
    "apiVersion": "horizon.openstack.org/v1alpha1",
    "blockOwnerDeletion": true,
    "controller": true,
    "kind": "Horizon",
    "name": "horizon",
    "uid": "48addfa8-3e8b-44f8-8d41-388e1eb8d00f"
  }
]
```

Reply:
Leaving it because it's something that was raised while discussing the approach for this. And there may be some established standard to lookup CR's by the label. So I'll take wider feedback on the topic.